### ISSUE TITLE
Don't send notifications about phases which users don't have access to

### ISSUE THREAD

**Issue by maxceem** [appirio-tech/connect-app#2972]
Customer users don't have access to the phases which are in draft state.

But if we create posts inside such phases or edit phase timelines and so on, customer users still get notification about new posts with links which lead to the phase which they cannot see.

Also, the customer can see notifications regarding the phase timeline and so on.

![image](https://user-images.githubusercontent.com/146016/56482091-e0edf600-64f4-11e9-8930-b6cd6a1adb52.png)

I guess customers shouldn't get notifications about the phases they don't have access to.

**Comment by vikasrohit** [comment]
I think we did something to prevent notifications for non visible phases to the customers. https://github.com/topcoder-platform/tc-notifications/issues/85. May be we just need to update the config now to exclude the messaging notifications of such phases.

If @maxceem you think you can get it done today by config changes, lets do that otherwise let us take it in next release.

**Comment by vikasrohit** [comment]
@maxceem @RishiRajSahu  Moving to next release as mostly no one post a message in a Draft phase.

**Comment by vikasrohit** [comment]
@maxceem this one demands dedicated handling instead of bug bash. If you think the same, you can get started with task for it.

**Comment by maxceem** [comment]
@vikasrohit yep, will start one.

**Comment by vikasrohit** [comment]
Great!! Please try to combine couple of more similar issues which we have for `tc-notifications`

**Comment by maxceem** [comment]
Just an update with details.

There are two cases which have to be handled:
1. Don't show notifications regarding posts inside phases in `draft` status to customers. This should be handled in `tc-notifcations` service.
2. Don't show other notifications related to `draft` phases to customers. These type of cases should be handled in `tc-project-service`.

**Comment by maxceem** [comment]
@vikasrohit The first part (1.) is fixed in https://github.com/topcoder-platform/tc-notifications/pull/117

Regarding the second part, I should first verify which notifications are still come to customers for draft phases and which don't. As there are already some filters implemented in project service as mentioned above.

**Comment by vikasrohit** [comment]
@maxceem I think as a customer I am not receiving notifications for phases which are still in planned state. However, I am not sure  if before this fix, were customer's receiving notification emails for the same as well?
Another thing, I noticed, is that another manager in the same project also didn't receive any notification (web/email) when I, as manager, added a new phase and updated phase's title. May be it is is intended behaviour as per current configuration, just wanted to note it down.
Closing the issue though as its fix serves the purpose.
